loopy/type_inference.py

```diff
@@ -216,8 +216,12 @@ def map_constant(self, expr):
             # Numpy types are sized
             return [NumpyType(np.dtype(type(expr)))]
         elif dt.kind == "f":
-            # deduce the smaller type by default
-            return [NumpyType(np.dtype(np.float32))]
+            if np.float32(expr) == np.float64(expr):
+                # No precision is lost by 'guessing' single precision, use that.
+                # This at least covers simple cases like '1j'.
+                return [NumpyType(np.dtype(np.float32))]
+
+            return [NumpyType(np.dtype(np.float64))]
         elif dt.kind == "c":
             if np.complex64(expr) == np.complex128(expr):
```

Comment:
I agree with the change here for inferring f32 or f64. But, why didn't we do a similar for the complex constants i.e. complex 64 or complex 128.

Reply:
If I understand your question correctly, what you're asking is, "why not do the same thing for complex as this proposes for float?"

Reply:
Bumped that discussion to #190, so that we can merge this and unbreak pytential. Hope that's OK with you.

Reply:
Yep, let's merge this.